### DISCUSSION TITLE
"Create New" option filled entire screen

### DISCUSSION THREAD

**Discussion by bps86** [lxqt/lxqt#2481] [Q&A] [answered]
So after using pcmanfm-qt after 4-5 days, there's this one feature(or maybe bug) where when I right click file/folder, hover at the "Create New", below Folder and Blank File, there's a huge amount of file that was not supposed to be in there. It literally fills my entire screen(I'm using i3wm) when I accidentally hover it and it's pretty annoying. Idk if it's related to this https://github.com/lxqt/pcmanfm-qt/pull/1490 because I'm using 1.2.1 . If it is, how do I remove the recent files feature below those two Folder and Blank File?

**Comment by tsujan** [comment]
> ... was not supposed to be in there...

They *are* supposed to be there — as in any other app that shows recent files.

You could disable it or reduce their number in Preferences → Behavior → Number of recent files.

**Reply by bps86** [reply]
That's the weird part. The number was set to 0 but it still shows, which made me curious. Was this a bug befor 1.3? If so I'll try to source build it and see what happens.

**Comment by stefonarch** [comment]
> when I accidentally hover

If I got it right this shows "templates" folder content. Check in session settings where it points to.

**Reply by bps86** [reply]
I don't think that's the case(template). The file was all over the place/random and I don't have template folder
Edit: Turns out it was the problem. Thanks for the help,

**Reply by bps86** [reply]
Here's my pcmanfm-qt settings
```
[Behavior]
AutoSelectionDelay=600
BookmarkOpenMethod=current_tab
ConfirmDelete=true
ConfirmTrash=false
CtrlRightClick=false
NoUsbTrash=false
QuickExec=false
RecentFilesNumber=0
SelectNewFiles=false
SingleClick=false
SingleWindowMode=false
UseTrash=true

[Desktop]
AllSticky=false
BgColor=#000000
DesktopCellMargins=@Size(3 1)
DesktopIconSize=48
DesktopShortcuts=@Invalid()
FgColor=#ffffff
Font="Sans Serif,9,-1,5,50,0,0,0,0,0"
HideItems=false
LastSlide=
OpenWithDefaultFileManager=false
PerScreenWallpaper=false
ShadowColor=#000000
ShowHidden=false
SlideShowInterval=0
SortColumn=name
SortFolderFirst=true
SortHiddenLast=false
SortOrder=ascending
TransformWallpaper=false
Wallpaper=
WallpaperDialogSize=@Size(700 500)
WallpaperDialogSplitterPos=200
WallpaperDirectory=
WallpaperMode=none
WallpaperRandomize=false
WorkAreaMargins=12, 12, 12, 12

[FolderView]
BackupAsHidden=false
BigIconSize=48
CustomColumnWidths=@Invalid()
FolderViewCellMargins=@Size(3 3)
HiddenColumns=@Invalid()
Mode=detailed
NoItemTooltip=false
ScrollPerPixel=false
ShadowHidden=true
ShowFilter=false
ShowFullNames=true
ShowHidden=true
SidePaneIconSize=24
SmallIconSize=24
SortCaseSensitive=false
SortColumn=type
SortFolderFirst=true
SortHiddenLast=false
SortOrder=ascending
ThumbnailIconSize=128

[Places]
HiddenPlaces=@Invalid()

[Search]
ContentPatterns=@Invalid()
MaxSearchHistory=0
NamePatterns=@Invalid()
searchContentCaseInsensitive=false
searchContentRegexp=true
searchNameCaseInsensitive=false
searchNameRegexp=true
searchRecursive=false
searchhHidden=false

[System]
Archiver=file-roller
FallbackIconThemeName=oxygen
OnlyUserTemplates=false
SIUnit=false
SuCommand=lxqt-sudo %s
TemplateRunApp=false
TemplateTypeOnce=false
Terminal=xterm

[Thumbnail]
MaxExternalThumbnailFileSize=-1
MaxThumbnailFileSize=4096
ShowThumbnails=true
ThumbnailLocalFilesOnly=false

[Volume]
AutoRun=true
CloseOnUnmount=true
MountOnStartup=true
MountRemovable=true

[Window]
AlwaysShowTabs=true
FixedHeight=480
FixedWidth=640
LastWindowHeight=703
LastWindowMaximized=false
LastWindowWidth=879
PathBarButtons=true
RememberWindowSize=true
ReopenLastTabs=false
ShowMenuBar=true
ShowTabClose=true
SidePaneMode=places
SidePaneVisible=true
SplitView=false
SplitterPos=150
SwitchToNewTab=false
TabPaths=@Invalid()
```

```
[Recent]
Files=@Invalid()
```

**Comment by Chiitoo** [comment]
> If I got it right this shows "templates" folder content. Check in session settings where it points to.

Heh, been meaning to look into this for quite a while now, since my 'create new' has been showing all the files under my home directory for a looong time.  :]

That would explain that.

Thanks!

**Reply by bps86** [reply]
wait, you're right it was all file from home dir. Did you fix it? If so, how?

**Reply by Chiitoo** [reply]
I suppose one could create an empty directory and point the templates session variable to it (that's what I'm going to try at least).

Not sure if there's a better good way to not need a directory for it at all.

**Reply by bps86** [reply]
Is this template session variable just for lxqt session manager only or is it actually global env variable? Sorry, I'm new at the session most part + quite ambigous since english's not my native language.

**Reply by isf63** [reply, marked as answer]
The XDG User Directories environment variables are global and used by nearly all desktop environments. Unfortunately XDG User Directories doesn't allow its directories to be set to nothing. They are normally set to folders immediately inside $HOME (~). For example, I don't use 'Public Share' so left it blank - however because XDG User Directories must be set with something, LXQt set it to $HOME.

It may be that the templates you see are coming from $HOME. I would create a folder ~/Templates with nothing in it and set LXQt Session to that.

**Reply by tsujan** [reply]
The file `~/.config/user-dirs.dirs` here:

```
# This file is written by xdg-user-dirs-update
# If you want to change or add directories, just edit the line you're
# interested in. All local changes will be retained on the next run
# Format is XDG_xxx_DIR="$HOME/yyy", where yyy is a shell-escaped
# homedir-relative path, or XDG_xxx_DIR="/yyy", where /yyy is an
# absolute path. No other format is supported.
# 
XDG_DESKTOP_DIR="$HOME/Desktop"
XDG_DOWNLOAD_DIR="$HOME/Downloads"
XDG_TEMPLATES_DIR="$HOME/Templates"
XDG_PUBLICSHARE_DIR="$HOME/Public"
XDG_DOCUMENTS_DIR="$HOME/Documents"
XDG_MUSIC_DIR="$HOME/Music"
XDG_PICTURES_DIR="$HOME/Pictures"
XDG_VIDEOS_DIR="$HOME/Videos"
```

It was there since I installed Manjaro-KDE in 2015, before migrating to LXQt a year later.

`XDG_TEMPLATES_DIR` has an important functionality, which all file managers use: templates for creating files of any kind. For example, I need a Lyx document to start writing math in it. So, I have `~/Templates/LyX-doc.lyx` as

```latex
#LyX 2.3 created this file. For more info see http://www.lyx.org/
\lyxformat 544
\begin_document
\begin_header
\save_transient_properties true
\origin unavailable
\textclass article
\begin_preamble
\usepackage{pgfplots}
\definecolor{darkblue}{RGB}{0, 0, 128}
\end_preamble
\use_default_options true
# and several more lines
```

Of course, `XDG_TEMPLATES_DIR` shouldn't be set to your home. And as @isf63 said, it's for all DEs.

**Reply by bps86** [reply]
Oh now it worked! Thanks guys, hopefully this solution will be available to public,

**Reply by tsujan** [reply]
Also, read https://wiki.archlinux.org/title/XDG_user_directories for a simple explanation.

**Reply by bps86** [reply]
Sure, Thanks for the reference,

**Reply by stefonarch** [reply]
I didn't like to have the directory `~/Templates` in my home so I created a folder in `~/Documents` for that.

**Reply by Chiitoo** [reply]
Indeed.

The LXQt Session Settings might be slightly unintuitive since in the User Directories the text boxes for the paths does have an 'x' button to clear it, and it will be happy to take that as a value, but it seems to always replace the empty ones with '$HOME' (probably what happened to me and/or forgot about it).

It seems to also take completely invalid input, such as '123' as well, though there is a warning about it (and then it returns to the previous value after closing).

**Reply by tsujan** [reply]
> I didn't like to have the directory ~/Templates

Perhaps you forgot that you could hide it: https://github.com/lxqt/pcmanfm-qt/wiki#hiding-filesfolders-without-renaming-them

As for me, I use `~/Templates` on a regular basis and put my temporary documents in `~/Documents` — to decide what I might want to do with them later.
